server/chat-plugins/room-events.ts

```diff
@@ -55,30 +72,61 @@ export const commands: ChatCommands = {
 				return this.errorReply("You're missing a command parameter - see /help roomevents for this command's syntax.");
 			}
 
-			const eventNameActual = eventName.trim();
+			let eventNameActual = eventName.trim();
 			const dateActual = date.trim();
 			const descString = desc.join(target.includes('|') ? '|' : ',').trim();
 
 			if (eventNameActual.length > 50) return this.errorReply("Event names should not exceed 50 characters.");
 			if (dateActual.length > 150) return this.errorReply("Event dates should not exceed 150 characters.");
 			if (descString.length > 1000) return this.errorReply("Event descriptions should not exceed 1000 characters.");
 
-			const eventId = toID(eventName);
+			const eventId = getEventIDByNameOrAlias(eventNameActual, room);
 			if (!eventId) return this.errorReply("Event names must contain at least one alphanumerical character.");
 
-			this.privateModAction(`(${user.name} ${room.events[eventId] ? "edited the" : "added a"} roomevent titled "${eventNameActual}".)`);
+			eventNameActual = (room.events[eventId] ? room.events[eventId].eventName : eventNameActual);
+			this.privateModAction(`(${user.name} ${room.events[eventId] ? "edited the" : "added a"} roomevent "${eventNameActual}".)`);
 			this.modlog('ROOMEVENT', null, `${room.events[eventId] ? "edited" : "added"} "${eventNameActual}"`);
 			room.events[eventId] = {
 				eventName: eventNameActual,
 				date: dateActual,
 				desc: descString,
 				started: false,
+				aliases: (room.events[eventId] ? room.events[eventId].aliases : []),
 			};
 
 			room.chatRoomData.events = room.events;
 			Rooms.global.writeChatRoomData();
 		},
 
+		rename(target, room, user) {
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
+			if (!this.can('ban', null, room)) return false;
+			let [oldName, newName] = target.split(target.includes('|') ? '|' : ',');
+			if (!(oldName && newName)) return this.errorReply("Usage: /roomevents rename [old name], [new name]");
+
+			newName = newName.trim();
+			const newID = toID(newName);
+			const oldID = (getAllAliases(room).includes(toID(oldName)) ? getEventIDByNameOrAlias(oldName, room) : toID(oldName));
+			if (newID === oldID) return this.errorReply("The new name must be different from the old one.");
+			if (!newID) return this.errorReply("Event names must contain at least one alphanumerical character.");
+			if (newName.length > 50) return this.errorReply("Event names should not exceed 50 characters.");
+
+			const eventData = room.events[oldID];
+			if (!eventData) return this.errorReply(`There is no event named '${oldName}'.`);
```

Comment:
```suggestion
			if (!eventData) return this.errorReply(`There is no event titled "${oldName}".`);
```

Just to keep consistent with the quotation usage elsewhere